src/sql_browser/async_std.rs

```diff
@@ -13,52 +13,50 @@ impl SqlBrowser for net::TcpStream {
+        let addrs = builder.get_addr().to_socket_addrs().await?;
+
+        for mut addr in addrs {
+            if let Some(ref instance_name) = builder.instance_name {
+                // First resolve the instance to a port via the
+                // SSRP protocol/MS-SQLR protocol [1]
+                // [1] https://msdn.microsoft.com/en-us/library/cc219703.aspx
+
+                let local_bind: std::net::SocketAddr = if addr.is_ipv4() {
+                    "0.0.0.0:0".parse().unwrap()
+                } else {
+                    "[::]:0".parse().unwrap()
+                };
+
+                let msg = [&[4u8], instance_name.as_bytes()].concat();
+                let mut buf = vec![0u8; 4096];
+
+                let socket = net::UdpSocket::bind(&local_bind).await?;
+                socket.send_to(&msg, &addr).await?;
+
+                let timeout = time::Duration::from_millis(1000);
+
+                let len = io::timeout(timeout, socket.recv(&mut buf))
+                    .map_err(|_| {
+                        crate::error::Error::Conversion(
+                            format!(
+                                "SQL browser timeout during resolving instance {}",
+                                instance_name
+                            )
+                            .into(),
+                        )
+                    })
+                    .await?;
 
-            let local_bind: std::net::SocketAddr = if addr.is_ipv4() {
-                "0.0.0.0:0".parse().unwrap()
-            } else {
-                "[::]:0".parse().unwrap()
+                let port = super::get_port_from_sql_browser_reply(buf, len, instance_name)?;
+                addr.set_port(port);
             };
 
-            let msg = [&[4u8], instance_name.as_bytes()].concat();
-            let mut buf = vec![0u8; 4096];
-
-            let socket = net::UdpSocket::bind(&local_bind).await?;
-            socket.send_to(&msg, &addr).await?;
-
-            let timeout = time::Duration::from_millis(1000);
-
-            let len = io::timeout(timeout, socket.recv(&mut buf))
-                .map_err(|_| {
-                    crate::error::Error::Conversion(
-                        format!(
-                            "SQL browser timeout during resolving instance {}",
-                            instance_name
-                        )
-                        .into(),
-                    )
-                })
-                .await?;
-
-            let port = super::get_port_from_sql_browser_reply(buf, len, instance_name)?;
-            addr.set_port(port);
-        };
+            if let Ok(stream) = net::TcpStream::connect(addr).await {
+                stream.set_nodelay(true)?;
+                return Ok(stream);
+            }
+        }
 
-        let stream = net::TcpStream::connect(addr).await?;
-        stream.set_nodelay(true)?;
-        Ok(stream)
+        Err(io::Error::new(io::ErrorKind::NotFound, "Could not resolve server host.").into())
```

Comment:
so here most of the diff is from indentation from putting it into a loop, not actual changes, right?